src/lib/ext-activitylog.js

```diff
@@ -65,9 +112,28 @@ class Controller {
     const existingLogs = await this.getExistingLogs();
 
     if (existingLogs.length) {
-      this.model.addNewLogs(existingLogs);
-      this.view.addTableRows(this.model.logs);
+      this.handleNewLogs(existingLogs);
+    }
+  }
+
+  handleNewLogs(logs) {
+    for (const log of logs) {
+      this.model.addNewLog(log);
+      this.isFilterMatched(this.model.filter, log)
```

Comment:
The visibility of the row currently follows two different code paths:
- here, when the log is added.
- when the filter is changed.

The logic is slightly different. Doing similar things in different ways is a recipe for bugs in the future, especially when unit tests are lacking.

Instead of explicitly passing the filter function whenever the log needs to be checked again, could you assign a filter function to the `view` upon its construction? Then the view can use this function when a row is added or when the filter is changed (i.e. which implies that the visibility of the rows need to be recomputed). This also enables you to keep the `addTableRows` logic from before (to continue to take advantage of DocumentFragment when appending rows in bulk).

Reply:
Thanks for explaining. 
In current update, `handleTableRows({ newLogs, filterObj })` method from `view` is fired when new logs are added or filter is changed.